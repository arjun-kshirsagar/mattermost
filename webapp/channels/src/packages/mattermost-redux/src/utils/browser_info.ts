// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export function getBrowserInfo() {
    const userAgent = window.navigator.userAgent;

    let browser = 'Unknown';
    let browserVersion = 'Unknown';

    // Simple browser detection
    if (userAgent.includes('Firefox')) {
        browser = 'Firefox';
    } else if (userAgent.includes('Chrome')) {
        browser = 'Chrome';
    } else if (userAgent.includes('Safari')) {
        browser = 'Safari';
    } else if (userAgent.includes('Edge')) {
        browser = 'Edge';
    }

    // Get browser version
    const match = userAgent.match(/(firefox|chrome|safari|edge(?=\/))\/?\s*(\d+)/i);
    if (match) {
        browserVersion = match[2];
    }

    return {browser, browserVersion};
}

export function getPlatformInfo() {
    // Casting to undefined in case it is deprecated in any browser
    const platform = window.navigator.platform as string | undefined;

    let platformName = 'Unknown';
    if (platform?.toLowerCase().includes('win')) {
        platformName = 'Windows';
    } else if (platform?.toLowerCase().includes('mac')) {
        platformName = 'MacOS';
    } else if (platform?.toLowerCase().includes('linux')) {
        platformName = 'Linux';
    }

    return platformName;
}
