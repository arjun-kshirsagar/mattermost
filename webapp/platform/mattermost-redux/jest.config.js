// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

/** @type {import('jest').Config} */

module.exports = {
    clearMocks: true,
    moduleNameMapper: {
        '^@mattermost/types/(.*)$': '<rootDir>/../types/src/$1',
        '^mattermost-redux/(.*)$': '<rootDir>/src/$1',
        '^.+\\.(jpg|jpeg|png|apng|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
            'identity-obj-proxy',
        '^.+\\.(css|less|scss)$': 'identity-obj-proxy',
        '^.*i18n.*\\.(json)$': '<rootDir>/../../channels/src/tests/i18n_mock.json',
    },
    // setupFilesAfterEnv: ['<rootDir>/test/setup_jest.ts'],
    // testEnvironment: 'jsdom',
};
