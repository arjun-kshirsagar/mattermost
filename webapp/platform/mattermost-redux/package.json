{
  "name": "mattermost-redux",
  "version": "9.3.0",
  "description": "Redux utilities for working with Mattermost",
  "keywords": [
    "mattermost"
  ],
  "homepage": "https://github.com/mattermost/mattermost/tree/master/webapp/platform/mattermost-redux#readme",
  "license": "MIT",
  "files": [
    "lib"
  ],
  "exports": {
    "./*": "./lib/*"
  },
  "typesVersions": {
    ">=3.1": {
      "*": [
        "./lib/*",
        "./lib/index.*"
      ]
    }
  },
  "repository": {
    "type": "git",
    "url": "github:mattermost/mattermost",
    "directory": "webapp/platform/mattermost-redux"
  },
  "dependencies": {
    "lodash": "^4.17.21",
    "moment-timezone": "^0.5.38",
    "redux": "^4.2.0",
    "redux-batched-actions": "^0.5.0",
    "redux-thunk": "^2.4.2",
    "rudder-sdk-js": "^1.0.16",
    "shallow-equals": "^1.0.0",
    "timezones.json": "^1.6.1"
  },
  "devDependencies": {
    "@types/jest": "28.1.8",
    "identity-obj-proxy": "3.0.0",
    "isomorphic-fetch": "3.0.0",
    "jest": "27.1.0",
    "jest-environment-jsdom": "29.7.0",
    "nock": "13.2.8",
    "@redux-devtools/extension": "3.2.3",
    "redux-mock-store": "1.5.4",
    "typescript": "^5.0.0"
  },
  "peerDependencies": {
    "@mattermost/types": "9.3.0",
    "@mattermost/client": "9.3.0",
    "typescript": "^4.3.0 || ^5.0.0"
  },
  "peerDependenciesMeta": {
    "typescript": {
      "optional": true
    }
  },
  "scripts": {
    "build": "tsc --build tsconfig.build.json --verbose",
    "check": "eslint --ext .js,.jsx,.tsx,.ts ./src --quiet",
    "run": "tsc --watch --preserveWatchOutput",
    "test": "cross-env TZ=Etc/UTC jest",
    "test-ci": "cross-env TZ=Etc/UTC jest --ci --forceExit --detectOpenHandles --maxWorkers=100%",
    "clean": "rm -rf lib node_modules *.tsbuildinfo"
  }
}
