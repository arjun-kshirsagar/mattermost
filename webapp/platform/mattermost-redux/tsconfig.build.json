{
    "compilerOptions": {
        "module": "commonjs",
        "moduleResolution": "node",
        "target": "es2022",
        "declaration": true,
        "strict": true,
        "resolveJsonModule": true,
        "isolatedModules": true,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "jsx": "react",
        "outDir": "./lib",
        "rootDir": "./src",
        "composite": true,
        "useUnknownInCatchVariables": false,
        "paths": {
            "mattermost-redux/*": ["./src/*"]
        }
    },
    "include": [
        "./src/**/*"
    ],
    "exclude": [
        "**/*.test.*"
    ],
    "references": [
        {"path": "../client"},
        {"path": "../types"}
    ]
}
