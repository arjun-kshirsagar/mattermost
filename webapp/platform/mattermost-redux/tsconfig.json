{
    "compilerOptions": {
        "allowJs": true,
        "module": "commonjs",
        "moduleResolution": "node",
        "target": "es2022",
        "declaration": true,
        "strict": true,
        "resolveJsonModule": true,
        "noEmit": true,
        "isolatedModules": true,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "jsx": "react",
        "outDir": "./lib",
        "rootDir": "./src",
        "composite": true,
        "useUnknownInCatchVariables": false,
        "paths": {
            "mattermost-redux/*": ["./src/*"]
        }
    },
    "include": [
        "./src/**/*"
    ],
    "references": [
        {"path": "../client"},
        {"path": "../types"}
    ]
}
