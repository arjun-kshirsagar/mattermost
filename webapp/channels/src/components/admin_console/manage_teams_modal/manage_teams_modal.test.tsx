// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {render, screen, waitFor} from '@testing-library/react';
import {act} from 'react-dom/test-utils';

import {General} from 'mattermost-redux/constants';

import ManageTeamsModal from 'components/admin_console/manage_teams_modal/manage_teams_modal';
import {renderWithContext} from 'tests/react_testing_utils';
import {TestHelper} from 'utils/test_helper';

// Mock the ManageTeamsDropdown component to avoid testing its internal logic
jest.mock('./manage_teams_dropdown', () => {
    return jest.fn((props) => {
        // Store props on mock function for later assertions
        (jest.fn as any).mockDropdownProps = props;
        return <div data-testid="mock-manage-teams-dropdown" />;
    });
});

describe('ManageTeamsModal', () => {
    const baseProps = {
        locale: General.DEFAULT_LOCALE,
        onHide: jest.fn(),
        onExited: jest.fn(),
        user: TestHelper.getUserMock({
            id: 'currentUserId',
            last_picture_update: 1234,
            email: 'currentUser@test.com',
            roles: 'system_user',
            username: 'currentUsername',
        }),
        actions: {
            getTeamMembersForUser: jest.fn().mockReturnValue(Promise.resolve({data: []})),
            getTeamsForUser: jest.fn().mockReturnValue(Promise.resolve({data: []})),
            updateTeamMemberSchemeRoles: jest.fn(),
            removeUserFromTeam: jest.fn(),
        },
    };

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('should match snapshot init', async () => {
        let container;
        await act(async () => {
            const result = renderWithContext(<ManageTeamsModal {...baseProps}/>);
            container = result.container;
        });
        expect(container).toMatchSnapshot();
    });

    test('should call api calls on mount', async () => {
        await act(async () => {
            renderWithContext(<ManageTeamsModal {...baseProps}/>);
        });

        expect(baseProps.actions.getTeamMembersForUser).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.getTeamMembersForUser).toHaveBeenCalledWith(baseProps.user.id);
        expect(baseProps.actions.getTeamsForUser).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.getTeamsForUser).toHaveBeenCalledWith(baseProps.user.id);
    });

    test('should render teams and team members from api calls', async () => {
        const mockTeamData = TestHelper.getTeamMock({
            id: '123test',
            name: 'testTeam',
            display_name: 'testTeam',
            delete_at: 0,
        });

        const getTeamMembersForUser = jest.fn().mockReturnValue(Promise.resolve({data: [{team_id: '123test'}]}));
        const getTeamsForUser = jest.fn().mockReturnValue(Promise.resolve({data: [mockTeamData]}));

        const props = {
            ...baseProps,
            actions: {
                ...baseProps.actions,
                getTeamMembersForUser,
                getTeamsForUser,
            },
        };

        await act(async () => {
            renderWithContext(<ManageTeamsModal {...props}/>);
        });

        // Wait for the team name to appear in the DOM
        await waitFor(() => {
            expect(screen.getByText('testTeam')).toBeInTheDocument();
        });

        // Check that the ManageTeamsDropdown received the correct props
        const ManageTeamsDropdown = require('./manage_teams_dropdown');
        expect(ManageTeamsDropdown).toHaveBeenCalled();
        expect((jest.fn as any).mockDropdownProps.teamMember).toEqual({team_id: '123test'});
    });
});
