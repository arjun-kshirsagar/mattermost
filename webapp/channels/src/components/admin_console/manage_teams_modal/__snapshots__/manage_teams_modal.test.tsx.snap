// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageTeamsModal should match snapshot init 1`] = `
<div
  aria-hidden="true"
/>
`;
